r/tests/testthat/test-compute-aggregate.R

```diff
@@ -37,7 +37,8 @@ test_that("sum.Array", {
 
   floats <- c(floats, NA)
   na <- Array$create(floats)
-  expect_identical(as.numeric(sum(na)), sum(floats))
+  # expect_identical(as.numeric(sum(na)), sum(floats))
+  expect_identical(as.numeric(sum(na)), NA_real_)
```

Comment:
Could also do something like (of course confirming the right way to determine a devel version). Could also add the devel check to the `skip_on_linux_cran` (and call it `skip_on_valgrind()`) if you wanted to make that narrower. 

```suggestion
  if (!grepl("devel", R.version.string)) {
    # Valgrind on R-devel confuses NaN and NA_real_
    expect_identical(as.numeric(sum(na)), sum(floats))
  }
```

Reply:
Yeah, I'll add the devel check to `skip_on_linux_cran()` (and probably rename it too, though I'm a bit hesitant to make it _so_ obvious that we are skipping the problematic tests — is that something we should worry about?).

As for these specific tests, looking at it now I wonder if we have benefit from `expect_identical(as.numeric(sum(na)), sum(floats))` versus `expect_identical(as.numeric(sum(na)), NA_real_)`? Is there a circumstance where we expect `sum(floats)` to change behavior and we would want to match it? I find `expect_identical(as.numeric(sum(na)), NA_real_)` to be easier to read at a glance and see exactly what it's testing/expecting.